### ISSUE TITLE
Set with array

### ISSUE THREAD

**Issue by henryiii** [scikit-hep/boost-histogram#189]
This seems to have managed to get closed by being with part of another issue, so adding it explicitly here.

We should support

```python
h[...] = array
```

Where `...` is any slice, including the actual `...`. If the array matches the size without flow, it fills without flow, and if it matches with flow, it fills the flow bins too. This was started in #165 with a draft version that only supports double storage. It is probably waiting on #186 to continue.

**Comment by henryiii** [comment]
This should be trivial to implement on top of #194, actually. Just needs a tiny bit of logic to support UHI locators. Very promising!

**Comment by HDembinski** [comment]
Nice feature, I like the automatic handling of flow.
